trade-finance-manager-ui/server/controllers/case/index.js

```diff
@@ -20,52 +20,57 @@ const {
 
-  const { user } = asUserSession(req.session);
+    const { user } = asUserSession(req.session);
 
-  const deal = await api.getDeal(dealId, userToken);
-  const { data: amendments } = await api.getAmendmentsByDealId(dealId, userToken);
+    const deal = await api.getDeal(dealId, userToken);
+    const { data: amendments } = await api.getAmendmentsByDealId(dealId, userToken);
 
-  if (!deal) {
-    return res.redirect('/not-found');
-  }
+    if (!deal) {
+      return res.redirect('/not-found');
+    }
 
-  if (!amendments || !Array.isArray(amendments)) {
-    console.error('Unable to get amendments for deal id %s', dealId);
-    return res.redirect('/not-found');
-  }
+    if (!amendments || !Array.isArray(amendments)) {
+      console.error('Unable to get amendments for deal id %s', dealId);
+      return res.redirect('/not-found');
+    }
 
-  const amendmentsInProgress = amendments.filter(({ status }) => status === AMENDMENT_STATUS.IN_PROGRESS);
-  const hasAmendmentInProgress = amendmentsInProgress.length > 0;
-  if (hasAmendmentInProgress) {
-    deal.tfm.stage = DEAL.DEAL_STAGE.AMENDMENT_IN_PROGRESS;
-  }
+    const amendmentsInProgress = amendments.filter(({ status }) => status === AMENDMENT_STATUS.IN_PROGRESS);
+    const hasAmendmentInProgress = amendmentsInProgress.length > 0;
+    if (hasAmendmentInProgress) {
+      deal.tfm.stage = DEAL.DEAL_STAGE.AMENDMENT_IN_PROGRESS;
+    }
 
-  const { submissionType } = deal.dealSnapshot;
+    const { submissionType } = deal.dealSnapshot;
 
-  const dealCancellationIsEnabled = isDealCancellationEnabled(submissionType, user);
-  let hasDraftCancellation = false;
+    const dealCancellationIsEnabled = isDealCancellationEnabled(submissionType, user);
+    let hasDraftCancellation = false;
 
-  if (dealCancellationIsEnabled) {
-    const cancellation = await api.getDealCancellation(dealId, userToken);
-    hasDraftCancellation = !isEmpty(cancellation);
-  }
+    if (dealCancellationIsEnabled) {
+      const cancellation = await api.getDealCancellation(dealId, userToken);
+      hasDraftCancellation = !isEmpty(cancellation);
+    }
 
-  return res.render('case/deal/deal.njk', {
-    deal: deal.dealSnapshot,
-    tfm: deal.tfm,
-    activePrimaryNavigation: 'manage work',
-    activeSubNavigation: 'deal',
-    dealId,
-    user: req.session.user,
-    amendments,
-    amendmentsInProgress,
-    hasAmendmentInProgress,
-    showDealCancelButton: dealCancellationIsEnabled,
-    hasDraftCancellation,
-  });
+    return res.render('case/deal/deal.njk', {
+      deal: deal.dealSnapshot,
+      tfm: deal.tfm,
+      activePrimaryNavigation: 'manage work',
+      activeSubNavigation: 'deal',
+      dealId,
+      user: req.session.user,
+      amendments,
+      amendmentsInProgress,
+      hasAmendmentInProgress,
+      showDealCancelButton: dealCancellationIsEnabled,
+      hasDraftCancellation,
+    });
+  } catch (error) {
+    console.error('Unable to render deal %o', error);
```

Comment:
Just a question, not a comment -- If we have an uncaught exception, it'll hit our middleware and show the problem with service page. It'll also log the error, which will include the stack trace through this function etc ie
```
err:
  controllers/case/index.js at getCaseDeal at LoC Bla
// goes further into depth about the stack trace
```

It seems the only thing this try catch is doing is appending `unable to render deal` -- Does this add value to debugging that justifies the additional indentations considering the stack trace will contain this information anyway?

Reply:
Good question, due to lack of exceptional handling this will cause `tfm-ui` microservice to crash thus requiring an auto reboot.

With above not only we gracefully handle any such exception i.e. `dealId` being invalid as parameter being provided is `undefined` but also helps pinpoints the issue and aide in quick debugging.